src/app/ErrorView/ErrorView.tsx

```diff
@@ -44,15 +44,31 @@ export interface ErrorViewProps {
 }
 
 export const ErrorView: React.FunctionComponent<ErrorViewProps> = (props) => {
-    return (<>
-        <br/>
-        <Bullseye>
-            <ExclamationCircleIcon size='md' color='Red'/>
-        </Bullseye>
-        <Bullseye>
-            <Text>
+
+    if (props.message === 'Target SSL Untrusted'){
```

Comment:
The message string shouldn't be used like this, since it's easy for that string to get reworded or otherwise modified (even returned in a different language/localization, for example). The HTTP status code from the response should be used to determine when this interaction is presented to the user, not the HTTP status message.

Reply:
I am trying to catch the SSL error, but I am wondering how I should handle the case of two handlers that return 502 codes related to Grafana. How can I destinguish between those and all of the rest?

Reply:
For that one handler that overloads the `502` code for both SSL trust failures and Grafana failures there is not (yet) a reliable way to distinguish those cases. Part of our API V2 work can/should be to come up with a better JSON API response format so that we can embed more detailed information within response bodies rather than relying solely upon HTTP status codes.

Since that's a much larger project, maybe the workaround for now is to remove the `502` overloading by changing the Grafana failure code to some other `5xx` code, and leave `502` for target SSL trust failures. This would be a much smaller PR on the backend that can go along with this one.